### ISSUE TITLE
Duplicated Code in Conditions

### ISSUE THREAD

**Issue by jubnzv** [nowarp/misti#87]
**Summary**
Duplicated code in conditional branches should be reported.

**Examples**

```tact
if (a > 42) {
  a = 43; // bad: duplicated code in both if and else branches
} else {
  a = 43;
}
```

Use instead:

```tact
if (a > 42) {
  a = inc(b); // ok
} else {
  a = 43;
}
```

**Comment by jubnzv** [comment]
Depends on https://github.com/tact-lang/tact/issues/314: API to compare AST entries.

**Comment by jubnzv** [comment]
Could be implemented, since Tact 1.4.4 introduced support of the basic AST equivalence API
